### PR TITLE
Bump android base modules to v0.0.1-alpha.3

### PR DESCRIPTION
**Pull request by LukasPaczos**
Refs https://github.com/mapbox/mapbox-base-android/releases/tag/v0.0.1-alpha.3.

Upgrades modules and removes the local `liblogger` in favor of the common implementation.

### CODE REVIEW

**Comment by RingerJK** [comment]
I think we should remove consumer proguard rules for `core`/`notification`/`router-s` modules https://github.com/mapbox/mapbox-navigation-android/blob/master/libnavigation-core/build.gradle#L27-L32

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt [hunk 1]
```diff
@@ -147,20 +147,19 @@ constructor(
...
+            MapboxModuleProvider.createModule(MapboxModuleType.NavigationRouter, ::paramsProvider)
         )
         directionsSession.registerRoutesObserver(internalRoutesObserver)
         directionsSession.registerRoutesObserver(navigationSession)
-        val notification: TripNotification = NavigationModuleProvider.createModule(
-            MapboxNavigationModuleType.TripNotification,
-            ::paramsProvider
-        )
+        val notification: TripNotification =
+            MapboxModuleProvider.createModule(
```
Comment (evabishchevich) [inline libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt:159]: NIT: formatting

**Comment by LukasPaczos** [comment]
@RingerJK unfortunately, we still need to prevent minification of Mapbox default modules that require arguments, see https://github.com/mapbox/mapbox-base-android/pull/35. We can keep using proguard rules for this, or `@Keep` annotation to just keep the constructor.

File: libnavigation-core/build.gradle [hunk 2]
```diff
@@ -41,6 +41,7 @@ android {
 
 dependencies {
     implementation dependenciesList.mapboxAnnotations
+    api dependenciesList.mapboxCommon
```
Comment (Guardiola31337) [inline libnavigation-core/build.gradle:44]: Does this need to be `api` so it's accessible from `libnavigation-ui`?

**Comment by LukasPaczos** [comment]
> Does this mean we can start adding back all the logging TODOs?

That's right! An instance of the logger should be created with the module provider and passed as an argument to all the modules that need logging capabilities.

File: libnavigation-core/build.gradle (hunk 2 above)
Comment (LukasPaczos) [reply libnavigation-core/build.gradle:44]: Yes, specifically for the `ReplayLocationEngine` that takes the logger as an argument. I'm unsure whether we should use a transitive dependency for this, or make UI lib properly instantiate the module.

Comment (Guardiola31337) [reply libnavigation-core/build.gradle:44]: > I'm unsure whether we should use a transitive dependency for this, or make UI lib properly instantiate the module.

That's right, probably in this case we can have both referenced on each module as `implementation` 🤔

Comment (LukasPaczos) [reply libnavigation-core/build.gradle:44]: On the other hand, we're adding navigation base (which also contains interface definitions) as `api`, so by adding common (which mostly serves the same purpose) as `api` we are consistent.

**Pull request merged**